test/Test.agda

```diff
@@ -1,40 +1,48 @@
-
 module _ where
 
 open import Agda.Builtin.List
 open import Agda.Builtin.Nat
+-- open import Agda.Builtin.Float
+-- open import Agda.Builtin.Word
+-- open import Agda.Builtin.Char
 open import Agda.Builtin.Equality
 
 variable
   a b : Set
 
+-- ** Foreign HS code
+
+-- language extensions
 {-# FOREIGN AGDA2HS
 {-# LANGUAGE LambdaCase #-}
 #-}
 
+-- imports
 {-# FOREIGN AGDA2HS
 import Prelude hiding (map, sum, (++))
 import Data.Monoid
+import Data.Word
 #-}
 
+-- ** Datatypes & functions
+
 data Exp (v : Set) : Set where
   Plus : Exp v → Exp v → Exp v
   Int : Nat → Exp v
   Var : v → Exp v
-
 {-# COMPILE AGDA2HS Exp #-}
 
 eval : (a → Nat) → Exp a → Nat
 eval env (Plus a b) = eval env a + eval env b
```

Comment:
The translation of `+` troubles me a bit, since it is only a coincidence that our Agda implementation of addition happens to be infix and called exactly the same as Haskell's addition on Nums. 
@UlfNorell somewhat related to the discussion on primitive functions